scripts/safeharbor/src/utils/contractUtils.js

```diff
@@ -0,0 +1,20 @@
+import { Contract, JsonRpcProvider } from "ethers";
+import { AGREEMENT_ADDRESS } from "../constants.js";
+import { AGREEMENTV2_ABI as AGREEMENT_ABI } from "../abis.js";
+
+// Create a provider instance
+export function createProvider() {
+    if (!process.env.ETH_RPC_URL) {
+        throw new Error("ETH_RPC_URL environment variable is not set");
+    }
+    return new JsonRpcProvider(process.env.ETH_RPC_URL);
+}
```

Comment:
I don't think it's a good idea to have environment variables scattered though several files.
You should care about them just at the top-level entrypoint.
For every other module, it's better to just stick with function parameters.

```suggestion
export function createProvider(ethRpcUrl) {
    if (!ethRpcUrl) {
        throw new Error("Invalid ethRpcUrl");
    }
    return new JsonRpcProvider(ethRpcUrl);
}
```

```suggestion
export function createProvider() {
    if (!process.env.ETH_RPC_URL) {
        throw new Error("ETH_RPC_URL environment variable is not set");
    }
    return new JsonRpcProvider(process.env.ETH_RPC_URL);
}
```